tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -506,8 +507,41 @@ class SPIRVEmitter : public ASTConsumer {
         return theBuilder.createLoad(resultType, fromValue);
       }
       return fromValue;
+    } else if (auto *cxxFunctionalCastExpr =
+                   dyn_cast<CXXFunctionalCastExpr>(expr)) {
+      // Explicit cast that is a NO-OP (e.g. vector<float, 4> -> float4)
+      if (cxxFunctionalCastExpr->getCastKind() == CK_NoOp) {
+        return doExpr(cxxFunctionalCastExpr->getSubExpr());
+      } else {
+        emitError("Found unhandled CXXFunctionalCastExpr cast type: %0")
+            << cxxFunctionalCastExpr->getCastKindName();
+      }
+    } else if (auto *initListExpr = dyn_cast<InitListExpr>(expr)) {
+      bool isConstantInitializer = expr->isConstantInitializer(
+          theCompilerInstance.getASTContext(), false);
+      const uint32_t resultType =
+          translateType(initListExpr->getType(), theBuilder);
+      std::vector<uint32_t> constituents;
+      for (size_t i = 0; i < initListExpr->getNumInits(); ++i) {
+        constituents.push_back(doExpr(initListExpr->getInit(i)));
+      }
+      if (isConstantInitializer) {
+        return theBuilder.getConstantComposite(resultType, constituents);
+      } else {
+        // TODO: use OpCompositeConstruct if it is not a constant initializer
+        // list.
+        emitWarning("Non-const initializer lists are currently not supported.");
+      }
+      // ehsan
+    } else if (auto *floatingLiteral = dyn_cast<FloatingLiteral>(expr)) {
+      const uint32_t resultType =
+          translateType(floatingLiteral->getType(), theBuilder);
+      const uint32_t value = floatingLiteral->getValue().convertToFloat();
```

Comment:
Wait... I  think `convertToFloat` returns a `float` value? We are converting it to `uint32_t` implicitly. This will lose precision.

We need test cases for non-`1` float values.